bittide/src/Bittide/DoubleBufferedRam.hs

```diff
@@ -150,93 +153,45 @@ wbStorage' initContent wbIn = delayControls wbIn wbOut
-
-  go WishboneM2S{..} rdB rdA romOut0 =
-    ( addrA
-    , addrB
-    , writeEntryA0
-    , writeEntryB0
-    , byteSelectA1
-    , byteSelectB1
-    , (emptyWishboneS2M @(Bytes 4)){acknowledge,readData,err})
+      (blockRamByteAddressableU, False)
+    NonReloadable content ->
+      ( blockRamByteAddressable @_ @depth content, False)
+
+  (readAddr, writeEntry, byteSelect, wbOut) =
+    unbundle (go <$> bundle  (wbIn, readData, romOut))
+
+  go (WishboneM2S{..}, readDataGo, (romWrite, romDone)) =
+    ( wbAddr
+    , writeEntryGo
+    , byteSelectGo
+    , (emptyWishboneS2M @(Bytes 4)){acknowledge,readData = readDataGo,err})
    where
-
-    (bitCoerce . resize -> wbAddr :: Index depth, alignment) =  split @_ @(aw - 2) addr
-
-    -- When the second lowest bit is not set, the address is considered word aligned.
-    -- We don't care about the first bit to determine the address alignment because
-    -- byte aligned addresses are considered illegal.
-    (not -> wordAligned, byteAligned) = bimap unpack unpack $ split alignment
-
-    -- The depth of the memory is defined as the number of words in the memory
-    -- (words are 4 bytes wide). The wishbone interface addresses per byte, so we multiply
-    -- the depth by 4 to get the number of bytes in the memory.
-    addrLegal = addr < (natToNum @(4 * depth)) && not byteAligned
+    (bitCoerce . resize -> wbAddr :: Index depth, alignment) = split @_ @(aw - 2) @2 addr
+    addrLegal = addr < (natToNum @(4 * depth)) && alignment == 0
 
     masterActive = strobe && busCycle
     err = masterActive && not addrLegal
+
     acknowledge = masterActive && (not isReloadable || romDone) && addrLegal
-    masterWriting = masterActive && writeEnable && not err
-
-    (bsHigh,bsLow) = split busSelect
-    (writeDataHigh, writeDataLow) = split writeData
-
-    ((addrA, byteSelectA0, writeDataA), (addrB, byteSelectB0, writeDataB))
-      | wordAligned =
-        ( (wbAddr, bsLow, writeDataLow)
-        , (wbAddr, bsHigh, writeDataHigh))
-      | otherwise =
-        ( (satSucc SatBound wbAddr, bsHigh, writeDataHigh)
-        , (wbAddr, bsLow, writeDataLow))
-
-    (romWrite, romDone) = romOut0
-    (romWriteB, romWriteA) = splitWrite romWrite
-    (writeEntryA0, writeEntryB0)
-      | isReloadable && not romDone = (romWriteA, romWriteB)
-      | masterWriting = (Just (addrA, writeDataA),Just (addrB, writeDataB))
-      | otherwise = (Nothing,Nothing)
-    readData
-      | wordAligned = rdB ++# rdA
-      | otherwise   = rdA ++# rdB
-
-    (byteSelectA1, byteSelectB1)
-      | isReloadable && not romDone = (maxBound,maxBound)
-      | otherwise = (byteSelectA0, byteSelectB0)
-
-  splitWrite ::
-    KnownNat bits =>
-    Maybe (LocatedBits n (2*bits)) ->
-    (Maybe (LocatedBits n bits), Maybe (LocatedBits n bits))
-  splitWrite (Just (i, a)) = (Just (i,upper), Just (i, lower))
-   where
-    (upper,lower) = split a
-  splitWrite Nothing = (Nothing, Nothing)
+
+    masterWriting = acknowledge && writeEnable
```

Comment:
Any reason the `&& not err` disappeared? I see it shouldn't really effect anything since if `err` then `! addrLegal`, so the write in `writeEntryGo` goes goes nowhere in the RAM, but I guess that could cause some issues since it's an invalid write?

Reply:
`acknowledge` implies `not err` :)

And you're right ,it would be a good idea to add tests for `wbStorageDP`.